wdl/transforms/draft3/src/main/scala/wdl/draft3/transforms/wdlom2wdl/WdlWriter.scala

```diff
@@ -0,0 +1,450 @@
+         |}""".stripMargin
+    }
+  }
+
+  implicit val taskDefinitionTypeElementWriter: WdlWriter[TaskDefinitionElement] = new WdlWriter[TaskDefinitionElement] {
+    override def toWdl(a: TaskDefinitionElement) = {
+      val inputs = a.inputsSection match {
+        case Some(i) => i.toWdl
+        case None => ""
+      }
+      val outputs = a.outputsSection match {
+        case Some(o) => o.toWdl
+        case None => ""
+      }
+      val runtime = a.runtimeSection match {
+        case Some(r) => r.toWdl
+        case None => ""
+      }
+      val meta = a.metaSection match {
+        case Some(m) => m.toWdl
+        case None => ""
+      }
+      val parameterMeta = a.parameterMetaSection match {
+        case Some(p) => p.toWdl
+        case None => ""
+      }
+
+      s"""task ${a.name} {
+         |${indent(inputs)}
+         |${indentAndCombine(a.declarations.map(_.toWdl))}
+         |${indent(outputs)}
+         |${a.commandSection.toWdl}
+         |${indent(runtime)}
+         |${indent(meta)}
+         |${indent(parameterMeta)}}""".stripMargin
+    }
+  }
+
+  implicit val commandSectionElementWriter: WdlWriter[CommandSectionElement] = new WdlWriter[CommandSectionElement] {
+    override def toWdl(a: CommandSectionElement): String = {
+      s"""command <<<
+         |${combine(a.parts.map(_.toWdl))}>>>""".stripMargin
+    }
+  }
+
+  implicit val commandSectionLineWriter: WdlWriter[CommandSectionLine] = new WdlWriter[CommandSectionLine] {
+    override def toWdl(a: CommandSectionLine): String = {
+      a.parts.map(_.toWdl).mkString
+    }
+  }
+
+  implicit val commandPartElementWriter: WdlWriter[CommandPartElement] = new WdlWriter[CommandPartElement] {
+    override def toWdl(a: CommandPartElement): String = a match {
+      case a: StringCommandPartElement => a.value // .trim?
+      case a: PlaceholderCommandPartElement =>
+        val attributes = a.attributes.toWdl
+
+        if (attributes.nonEmpty)
+          s"~{$attributes ${a.expressionElement.toWdl}}"
+        else
+          s"~{${a.expressionElement.toWdl}}"
+    }
+  }
+
+  implicit val placeholderAttributeSetWriter: WdlWriter[PlaceholderAttributeSet] = new WdlWriter[PlaceholderAttributeSet] {
+    override def toWdl(a: PlaceholderAttributeSet): String = {
+      Map(
+        "sep" -> a.sepAttribute,
+        "true" -> a.trueAttribute,
+        "false" -> a.falseAttribute,
+        "default" -> a.defaultAttribute
+      ).map({ case (attrKey: String, maybeValue: Option[String]) =>
+        maybeValue match {
+          case Some(value) => attrKey + "=\"" + value + "\""
+          case None => ""
+        }
+      }).filterNot(_.length == 0).mkString(sep = " ")
+    }
+  }
+
+  implicit val inputsSectionElementWriter: WdlWriter[InputsSectionElement] = new WdlWriter[InputsSectionElement] {
+    override def toWdl(a: InputsSectionElement): String = {
+      s"""input {
+         |${indentAndCombine(a.inputDeclarations.map(_.toWdl))}}""".stripMargin
```

Comment:
I think I'd prefer that closing `}` to be on a newline

Reply:
Are you referring to the source code or the output?

With the brace in its current position, the output is
```
  Array[Pair[String, Int]?] lotsa_nesting_array
}
```
and on its own line
```
  Array[Pair[String, Int]?] lotsa_nesting_array

}
```

Reply:
We could have it both ways by modifying how `combine` works, looking into that for consistency